packages/@ourworldindata/grapher/src/lineCharts/LineChart.tsx

```diff
@@ -300,6 +300,26 @@ export class LineChart
         return table
     }
 
+    transformTableForSelection(table: OwidTable): OwidTable {
+        // if entities with partial data are not plotted,
+        // make sure they don't show up in the entity selector
+        if (this.missingDataStrategy === MissingDataStrategy.hide) {
+            table = table.replaceNonNumericCellsWithErrorValues(
+                this.yColumnSlugs
+            )
+
+            const groupedByEntity = table.groupBy("entityName").map((t) => {
+                if (t.hasAnyColumnNoValidValue(this.yColumnSlugs)) {
+                    t = t.dropAllRows()
+                }
+                return t
+            })
+            table = groupedByEntity[0].concat(groupedByEntity.slice(1))
```

Comment:
This expects `groupedByEntity.length > 0`, which I'm not sure is always the case.

Reply:
I think it's very likely since all entities being removed would be an extreme edge case, but I can guard against it in any case...